llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -693,11 +694,11 @@ void LoopVectorizationLegality::addInductionPhi(
   const DataLayout &DL = Phi->getDataLayout();
 
   // Get the widest type.
-  if (!PhiTy->isFloatingPointTy()) {
+  if (PhiTy->isIntOrPtrTy()) {
```

Comment:
Looks like the original code was potentially also accepting structs, aggregates and arrays, etc. However, I can only assume that these would be rejected by LoopVectorizationLegality somewhere so I suppose that still makes it NFC.

Reply:
If it weren't rejected by LoopVectorizationLegality, the later casts in LoopVectorize (which are stripped in this patch) would have crashed, no?

Reply:
This should be fine, induction descriptor only supports integer, pointer and fp types.